### DISCUSSION TITLE
Achieve unique session based on Client id

### DISCUSSION THREAD

**Discussion by StanLi-ops** [ntex-rs/ntex-mqtt#77] [Q&A] [answered]
Hello, I hope to achieve the following scenarios:
`
Client A uses Client_id:'A' to connect to the broker.
Client B uses the same Client_id:'A' to connect, and the broker will disconnect the session with Client A and establish a new session with Client B.
`

I think it should be handled in the function handshake(). But I don't know how to close the original session in the function handshake().

**Comment by fafhrd91** [comment, marked as answer]
you need registry of available sessions. then you just take session for client and disconnect it
